eng/scripts/build.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from pathlib import Path
+from helpers import is_pipeline_build, call
+import argparse
+import shutil
+import json
+
+pipeline_build = is_pipeline_build()
+root_dir = Path.resolve(Path(__file__) / "../../..")
+
+def main(package_name: str, version_suffix: str, output_dir: str, package_versions: dict):
+    if package_name == "autorest.python":
+        versionVariableName = "generatorVersion"
+        tarball_prefix = "autorest-python"
+    elif package_name == "typespec-python":
+        versionVariableName = "emitterVersion"
+        tarball_prefix = "azure-tools-typespec-python"
+    else:
+        print(f"Invalid package name: {package_name}")
+        exit(1)
+
+    package_dir = root_dir / Path("packages") / package_name
+
+    currentVersion = call(
+        "node -pe \"require('./package.json').version\"", capture_output=True, cwd=package_dir
+    ).strip()
+
+    if version_suffix:
+        newVersion = f"{currentVersion}{version_suffix}"
+
+        # Update version
+        call(f'npm version "{newVersion}" --no-workspaces-update', cwd=package_dir)
+        call("git add package.json", cwd=package_dir)
+    else:
+        newVersion = currentVersion
+
+    print(f"Building version {newVersion}")
+
+    package_versions[package_name] = newVersion
+
+    if pipeline_build:
+        # Update version variable
+        print(
+            f"##vso[task.setvariable variable={versionVariableName};isOutput=true]{newVersion}"
+        )
+
+    # Build project
+    call("pnpm run build")
+
+    # Check version mismatch
+    call("pnpm check-version-mismatch")
```

Comment:
nit: do we also want to check formatting / cspell / linting? In [`cadl-ranch`](https://github.com/Azure/cadl-ranch/blob/main/package.json#L26), so we could just call `pnpm run pr-prep`. We could add this to our package.json, and also clean up code in the other templates as well

Reply:
this is a copy of the individual script calls in [ci-template.yaml](https://github.com/Azure/autorest.python/blob/2f14606c12e68c3c6172456ffba15706407f6c8a/eng/pipelines/ci-template.yml) with tests or code checks pulled out to test.py

Reply:
We should make a pass through the template and the scripts to make sure I've captured everything correctly and make changes like `pnpm run pr-prep`